compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -293,7 +295,91 @@ def _do_logout_endpoints(
     return tokens_revoked, error_msg
 
 
-def read_config(endpoint_dir: pathlib.Path) -> Config:
+FUNCX_COMPUTE_IMPORT_UPDATES = {
+    "from funcx_endpoint.endpoint.utils.config": "from globus_compute_endpoint.endpoint.utils.config",  # noqa E501
+    "from funcx_endpoint.executors": "from globus_compute_endpoint.executors",  # noqa E501
+}
+
+
+def _upgrade_funcx_imports_in_config(name: str, force=False) -> str:
+    """
+    This only modifies unindented import lines, as are in the original
+    config.py.  Indented matching lines are user created and would have to be
+    updated manually by the user.
+
+    The force flag will overwrite an existing (non-directory) config.py.bak
+
+    This method uses a randomly generated intermediate output file in case
+    there are any permission or unforeseen file system issues
+    """
+    ep_dir = get_config_dir() / name
+    old_config = ep_dir / "config.py"
+    config_backup = ep_dir / "config.py.bak"
+
+    try:
+        # Scan config.py first in case it's a no-op
+        with open(old_config) as f:
+            lines = f.readlines()
```

Comment:
These are all `pathlib.Path` objects, so you might consider utilizing their power, where applicable.  For example, you might replace this `with` statement with:
```
lines = old_config.read_text().splitlines(keepends=True)
```
(Although, have made a suggestion privately that the splitlines may not be necessary as well.)